apps/web/src/components/digest-demo-flow/DigestDemoFlowProvider.tsx

```diff
@@ -0,0 +1,121 @@
+
+const DigestDemoFlowContext = React.createContext({
+  isReadOnly: true,
+  isRunningDigest: true,
+  digestInterval: 10,
+  triggerCount: 0,
+  emailsSentCount: 0,
+  updateDigestInterval: (interval: number) => {},
+  runTrigger: () => {},
+});
+
+export const useDigestDemoFlowContext = () => useContext(DigestDemoFlowContext);
+
+export const DigestDemoFlowProvider = ({
+  children,
+  isReadOnly,
+  templateId,
+}: {
+  children: ReactNode;
+  isReadOnly: boolean;
+  templateId?: string;
+}) => {
+  const [{ isRunningDigest, triggerCount, digestInterval, emailsSentCount }, setState] = useState({
+    isRunningDigest: false,
+    triggerCount: 0,
+    digestInterval: 10,
+    emailsSentCount: 0,
+  });
+  const { currentUser } = useAuthContext();
+  const { template } = useTemplateFetcher(
+    { templateId },
+    {
+      enabled: !isReadOnly && !!templateId,
+      refetchOnWindowFocus: false,
+      onSuccess: (fetchedTemplate) => {
+        const digest: INotificationTemplateStep | undefined = fetchedTemplate.steps.find(
+          (step) => step.template?.type === StepTypeEnum.DIGEST
+        );
+        if (digest) {
+          setState((state) => ({ ...state, digestInterval: digest.metadata?.amount ?? 10 }));
+        }
+      },
+    }
+  );
+  const { updateTemplateMutation } = useUpdateTemplate();
+  const debouncedUpdateTemplate = useDebounce(
+    (args: { id: string; data: Partial<IUpdateNotificationTemplateDto> }) => updateTemplateMutation(args),
+    1000
+  );
+  const { mutateAsync: testTriggerMutation } = useMutation(testTrigger);
+
+  const updateDigestInterval = useCallback(
+    (interval: number) => {
+      setState((state) => ({ ...state, digestInterval: interval }));
+
+      if (template && template._id) {
+        const steps = template.steps.map((step) => {
+          if (step.template?.type === StepTypeEnum.DIGEST) {
+            return { ...step, metadata: { ...step.metadata, amount: interval } };
+          }
+
+          return step;
+        });
+
+        debouncedUpdateTemplate({ id: template._id, data: { steps } });
+      }
+    },
+    [debouncedUpdateTemplate, template]
+  );
+
+  const runTrigger = useCallback(() => {
+    if (template && currentUser) {
+      setState((state) => ({ ...state, triggerCount: state.triggerCount + 1 }));
+
+      testTriggerMutation({
+        name: template.triggers[0].identifier,
+        to: [currentUser._id],
+        payload: {},
+        overrides: {},
+      });
+    }
+  }, [template, currentUser]);
+
+  useEffect(() => {
+    if (template && !isRunningDigest && triggerCount > 0) {
+      setState((state) => ({ ...state, isRunningDigest: true }));
+
+      setTimeout(() => {
+        setState((state) => ({
+          ...state,
+          isRunningDigest: false,
+          triggerCount: 0,
+          emailsSentCount: state.emailsSentCount + 1,
+        }));
+      }, digestInterval * 1000);
+    }
+  }, [template, isRunningDigest, triggerCount, digestInterval]);
```

Comment:
this effect is used to run the digest timer... like when the user clicks the run trigger button we update the `triggerCount` and this effect is fired... then we say that the digest is running and we do fire here the timeout...
the callback clears the `triggerCount, isRunningDigest` and updates the number of emails sent count

Reply:
small suggestion, what do you think about creating a function that separates the logic inside so the if statement so it will be more readable in the first view?